apollo-router/src/services/layers/apq.rs

```diff
@@ -123,14 +119,14 @@ pub(crate) async fn apq_request(
             if query_matches_hash(query.as_str(), query_hash_bytes.as_slice()) {
                 tracing::trace!("apq: cache insert");
                 let _ = request.context.insert("persisted_query_hit", false);
-                cache.insert(format!("apq|{query_hash}"), query).await;
+                cache.insert(redis_key(&query_hash), query).await;
             } else {
                 tracing::warn!("apq: graphql request doesn't match provided sha256Hash");
             }
             Ok(request)
         }
         (Some((apq_hash, _)), _) => {
-            if let Ok(cached_query) = cache.get(&format!("apq|{apq_hash}")).await.get().await {
+            if let Ok(cached_query) = cache.get(&redis_key(&apq_hash)).await.get().await {
```

Comment:
That would be great to have something more typesafe, something that takes an `impl RedisKey` or something but we can improve this later

Reply:
in the end the redis key is a byte array so it's not a large issue here